### ISSUE TITLE
CRAN package check test suite fails for Fedora and Solaris

### ISSUE THREAD

**Issue by rcst** [rcst/rim#17]
<div class="container">
<h2>CRAN Package Check Results for Package <a href="https://cran.r-project.org/web/packages/rim/index.html"> rim </a> </h2>
<p>
Last updated on 2021-06-28 04:56:18 CEST.
</p>

Flavor | Version | Tinstall | Tcheck | Ttotal | Status | Flags
-- | -- | -- | -- | -- | -- | --
r-devel-linux-x86_64-debian-clang | 0.4.0 | 26.08 | 31.87 | 57.95 | OK |  
r-devel-linux-x86_64-debian-gcc | 0.4.0 | 15.05 | 25.17 | 40.22 | OK |  
r-devel-linux-x86_64-fedora-clang | 0.4.0 |   |   | 86.48 | ERROR |  
r-devel-linux-x86_64-fedora-gcc | 0.4.0 |   |   | 73.71 | ERROR |  
r-devel-windows-x86_64 | 0.4.0 | 29.00 | 60.00 | 89.00 | OK |  
r-devel-windows-x86_64-gcc10-UCRT | 0.4.0 |   |   |   | OK |  
r-patched-linux-x86_64 | 0.4.0 | 17.07 | 30.92 | 47.99 | OK |  
r-patched-solaris-x86 | 0.4.0 |   |   | 4.90 | ERROR |  
r-release-linux-x86_64 | 0.4.0 | 17.36 | 31.06 | 48.42 | OK |  
r-release-macos-x86_64 | 0.4.0 |   |   |   | OK |  
r-release-windows-ix86+x86_64 | 0.4.0 | 42.00 | 56.00 | 98.00 | OK |  
r-oldrel-macos-x86_64 | 0.4.0 |   |   |   | OK |  

<h3>Check Details</h3>

<p>

Version: 0.4.0

<br>

Check: tests

<br>

Result: ERROR

<br>

      Running ‘testthat.R’ [7s/10s]<br>
    Running the tests in ‘tests/testthat.R’ failed.<br>
    Complete output:<br>
      &gt; library(testthat)<br>
      &gt; test_check("rim")<br>
      Loading required package: rim<br>
      Maxima successfully registered as knitr engine!<br>
      Quitting from lines 16-18 (/data/gannet/ripley/R/packages/tests-clang/rim.Rcheck/rim/extdata/test.Rmd) <br>
      ══ Skipped tests ═══════════════════════════════════════════════════════════════<br>
      • No warnings to be tested under Maxima 5.43.0 <br>
       (1)<br>
      <br>
      ══ Failed tests ════════════════════════════════════════════════════════════════<br>
      ── Error (test-apropos.R:4:3): apropos returns formatted strings ───────────────<br>
      Error: Unsupported: <br>
      incorrect syntax: Premature termination of input at ;.<br>
      ;<br>
       ^<br>
      prompt;&gt;&gt;(%i3) &lt;&lt;prompt;<br>
      Backtrace:<br>
          █<br>
       1. ├─testthat::expect_match(maxima.apropos("int"), "^[[:space:]|[:print:]]*$") test-apropos.R:4:2<br>
       2. └─rim::maxima.apropos("int")<br>
       3.   └─maxima.env$maxima$get(paste0("apropos(\"", keystring, "\");"))<br>
      ── Error (test-ask-user.R:6:3): ask prompts are returned ───────────────────────<br>
      Error: Unsupported: <br>
      incorrect syntax: Premature termination of input at ;.<br>
      ;<br>
       ^<br>
      prompt;&gt;&gt;(%i4) &lt;&lt;prompt;<br>
      Backtrace:<br>
          █<br>
       1. ├─testthat::expect_match(maxima.get("y;"), "^[[:space:]|[:print:]]*$") test-ask-user.R:6:2<br>
       2. │ └─testthat::quasi_label(enquo(object), label, arg = "object")<br>
       3. │   └─rlang::eval_bare(expr, quo_get_env(quo))<br>
       4. └─rim::maxima.get("y;")<br>
       5.   └─maxima.env$maxima$get(command)<br>
      ── Error (test-engine.R:11:3): maxima knitr engine works ───────────────────────<br>
      Error: Unsupported: <br>
      incorrect syntax: Premature termination of input at ;.<br>
      ;<br>
       ^<br>
      prompt;&gt;&gt;(%i5) &lt;&lt;prompt;<br>
      Backtrace:<br>
           █<br>
        1. └─knitr::knit(input = fr[1], output = fo, quiet = TRUE, envir = .GlobalEnv) test-engine.R:11:2<br>
        2.   └─knitr:::process_file(text, output)<br>
        3.     ├─base::withCallingHandlers(...)<br>
        4.     ├─knitr:::process_group(group)<br>
        5.     └─knitr:::process_group.block(group)<br>
        6.       └─knitr:::call_block(x)<br>
        7.         └─knitr:::block_exec(params)<br>
        8.           ├─knitr:::in_dir(input_dir(), engine(options))<br>
        9.           └─rim:::engine(options)<br>
       10.             └─maxima.env$mx$get(code[i])<br>
      ── Error (test-execute.R:5:3): execution returns formatted strings ─────────────<br>
      Error: Unsupported: <br>
      incorrect syntax: Premature termination of input at ;.<br>
      ;<br>
       ^<br>
      prompt;&gt;&gt;(%i7) &lt;&lt;prompt;<br>
      Backtrace:<br>
          █<br>
       1. ├─testthat::expect_match(maxima.get("2+2;"), "^4$") test-execute.R:5:2<br>
       2. │ └─testthat::quasi_label(enquo(object), label, arg = "object")<br>
       3. │   └─rlang::eval_bare(expr, quo_get_env(quo))<br>
       4. └─rim::maxima.get("2+2;")<br>
       5.   └─maxima.env$maxima$get(command)<br>
      ── Error (test-setformat.R:8:3): changing the output format works ──────────────<br>
      Error: Unsupported: <br>
      incorrect syntax: Premature termination of input at ;.<br>
      ;<br>
       ^<br>
      prompt;&gt;&gt;(%i10) &lt;&lt;prompt;<br>
      Backtrace:<br>
          █<br>
       1. ├─testthat::expect_match(maxima.get("2+2;"), "4") test-setformat.R:8:2<br>
       2. │ └─testthat::quasi_label(enquo(object), label, arg = "object")<br>
       3. │   └─rlang::eval_bare(expr, quo_get_env(quo))<br>
       4. └─rim::maxima.get("2+2;")<br>
       5.   └─maxima.env$maxima$get(command)<br>
      <br>
      [ FAIL 5 | WARN 0 | SKIP 1 | PASS 4 ]<br>
      Error: Test failures<br>
      Execution halted

<br>

Flavor: <a href="https://www.r-project.org/nosvn/R.check/r-devel-linux-x86_64-fedora-clang/rim-00check.html">r-devel-linux-x86_64-fedora-clang</a>

</p>

<p>

Version: 0.4.0

<br>

Check: tests

<br>

Result: ERROR

<br>

      Running ‘testthat.R’ [9s/13s]<br>
    Running the tests in ‘tests/testthat.R’ failed.<br>
    Complete output:<br>
      &gt; library(testthat)<br>
      &gt; test_check("rim")<br>
      Loading required package: rim<br>
      Maxima successfully registered as knitr engine!<br>
      Quitting from lines 10-13 (/data/gannet/ripley/R/packages/tests-devel/rim.Rcheck/rim/extdata/test.Rmd) <br>
      ══ Skipped tests ═══════════════════════════════════════════════════════════════<br>
      • No warnings to be tested under Maxima 5.43.0 <br>
       (1)<br>
      <br>
      ══ Failed tests ════════════════════════════════════════════════════════════════<br>
      ── Error (test-ask-user.R:6:3): ask prompts are returned ───────────────────────<br>
      Error: Unsupported: <br>
      incorrect syntax: Premature termination of input at ;.<br>
      ;<br>
       ^<br>
      prompt;&gt;&gt;(%i4) &lt;&lt;prompt;<br>
      Backtrace:<br>
          █<br>
       1. ├─testthat::expect_match(maxima.get("y;"), "^[[:space:]|[:print:]]*$") test-ask-user.R:6:2<br>
       2. │ └─testthat::quasi_label(enquo(object), label, arg = "object")<br>
       3. │   └─rlang::eval_bare(expr, quo_get_env(quo))<br>
       4. └─rim::maxima.get("y;")<br>
       5.   └─maxima.env$maxima$get(command)<br>
      ── Error (test-engine.R:11:3): maxima knitr engine works ───────────────────────<br>
      Error: Unsupported: <br>
      incorrect syntax: Premature termination of input at ;.<br>
      ;<br>
       ^<br>
      prompt;&gt;&gt;(%i4) &lt;&lt;prompt;<br>
      Backtrace:<br>
           █<br>
        1. └─knitr::knit(input = fr[1], output = fo, quiet = TRUE, envir = .GlobalEnv) test-engine.R:11:2<br>
        2.   └─knitr:::process_file(text, output)<br>
        3.     ├─base::withCallingHandlers(...)<br>
        4.     ├─knitr:::process_group(group)<br>
        5.     └─knitr:::process_group.block(group)<br>
        6.       └─knitr:::call_block(x)<br>
        7.         └─knitr:::block_exec(params)<br>
        8.           ├─knitr:::in_dir(input_dir(), engine(options))<br>
        9.           └─rim:::engine(options)<br>
       10.             └─maxima.env$mx$get(code[i])<br>
      ── Error (test-execute.R:6:3): execution returns formatted strings ─────────────<br>
      Error: Unsupported: <br>
      incorrect syntax: Premature termination of input at ;.<br>
      ;<br>
       ^<br>
      prompt;&gt;&gt;(%i10) &lt;&lt;prompt;<br>
      Backtrace:<br>
          █<br>
       1. ├─testthat::expect_match(...) test-execute.R:6:2<br>
       2. └─rim::maxima.get("jacobian( [alpha / (alpha + beta), 1 / sqrt(alpha + beta)], [alpha, beta] )")<br>
       3.   └─maxima.env$maxima$get(command)<br>
      ── Error (test-setformat.R:8:3): changing the output format works ──────────────<br>
      Error: Unsupported: <br>
      incorrect syntax: Premature termination of input at ;.<br>
      ;<br>
       ^<br>
      prompt;&gt;&gt;(%i13) &lt;&lt;prompt;<br>
      Backtrace:<br>
          █<br>
       1. ├─testthat::expect_match(maxima.get("2+2;"), "4") test-setformat.R:8:2<br>
       2. │ └─testthat::quasi_label(enquo(object), label, arg = "object")<br>
       3. │   └─rlang::eval_bare(expr, quo_get_env(quo))<br>
       4. └─rim::maxima.get("2+2;")<br>
       5.   └─maxima.env$maxima$get(command)<br>
      <br>
      [ FAIL 4 | WARN 0 | SKIP 1 | PASS 6 ]<br>
      Error: Test failures<br>
      Execution halted

<br>

Flavor: <a href="https://www.r-project.org/nosvn/R.check/r-devel-linux-x86_64-fedora-gcc/rim-00check.html">r-devel-linux-x86_64-fedora-gcc</a>

</p>

<p>

Version: 0.4.0

<br>

Check: whether package can be installed

<br>

Result: ERROR

<br>

    Installation failed.

<br>

Flavor: <a href="https://www.r-project.org/nosvn/R.check/r-patched-solaris-x86/rim-00check.html">r-patched-solaris-x86</a>

</p>

</div>

**Comment by rcst** [comment]
After a first inspection using Fedora 34 in VM the errors seem to be caused by an unfortunate reading-split of lines.

On my for example I get a read out from the socket like this:
```
out;>>
(
%o3) log(x)
<<out;
prompt;>>(%i4) <<prompt;
```

Which then fails to read the output label being used as validation in `Reply$initialize(con)`

```
39: regex(text = [...], pattern = "out;>>\\s+\\((%o(\\d+))\\)\\s\\n?([[:space:]|[:print:]]*)\\s+<<out;")
```

**Comment by rcst** [comment]
fixed in `541d167`
